Tests/ArgumentParserUnitTests/HelpGenerationTests.swift

```diff
@@ -514,6 +514,54 @@ extension HelpGenerationTests {
     
     """)
   }
+
+  struct WithSubgroups: ParsableCommand {
+    static let configuration = CommandConfiguration(
+      commandName: "subgroupings"
+    ) {
+      CommandGroup(
+        name: "Broken",
+        abstract: "These commands are currently non-functional due to XYZ."
+      ) {
+        Foo.self
+        Bar.self
+      }
+
+      if true {
+        M.self
+      } else {
+        // nothing here
+      }
+
+      CommandGroup(name: "Complicated") {
+        if #available(macOS 10.10, *) {
+          N.self
+        }
+      }
+    }
+  }
+
+  func testHelpSubcommandGroups() throws {
+    AssertHelp(.default, for: WithSubgroups.self, columns: 50, equals: """
+    USAGE: subgroupings <subcommand>
+
+    OPTIONS:
+      -h, --help              Show help information.
+
+    SUBCOMMANDS:
+      m
+
+    BROKEN SUBCOMMANDS: These commands are currently non-functional due
+    to XYZ.
+      foo                     Perform some foo
+      bar                     Perform bar operations
+
+    COMPLICATED SUBCOMMANDS:
+      n
+
+      See 'subgroupings help <subcommand>' for detailed help.
+    """)
+  }
```

Comment:
This output looks terrific! From what I can see, empty command groups are allowed, but essentially ignored in the output, which is the right behavior since we're supporting conditionals in the builder syntax. Can we add a test with one of those to verify? Also a test with _only_ grouped subcommands, to make sure we aren't missing anything in that case.